### ISSUE TITLE
Please use git submofules or external packages instead of 'git clone' (for caffee)

### ISSUE THREAD

**Issue by yurivict** [jolibrain/deepdetect#585]
In the modern package builders projects aren't allowed to download anything for security reasons.

It's best to use the external packages. They are available for ```caffee``` for all major platforms, see here https://repology.org/project/caffe/versions

The next best things is to use git submodules: https://git-scm.com/book/en/v2/Git-Tools-Submodules

**Comment by fantes** [comment]
We use our own extremelly customized caffe version whis not distributed outside of github, so we will have a look at git submodules as soon as we have some time :)

**Comment by beniz** [comment]
We could use git submodules though. However this is similar to a download, and our downloads are from GitHub straight and this should be as safe (or not) as a git clone.

**Comment by yurivict** [comment]
>  this should be as safe (or not) as a git clone.

No, downloads and submodules aren't as safe as fingerprinted downloads that are used by package builders. (Fingerprinted downloads are exactly 100% reproducible, while regular https downloads are not.)

git submodules though just place extra code directories into the same source tree before the build, process. There is a command to do that in the package builders. You download the files during the build process into the build directory. this requires special patching, figuring out how to do this. This makes your project less likely to have ports for various distros, because it is just more difficult.
